env0/resource_template_project_assignment.go

```diff
@@ -0,0 +1,90 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTemplateProjectAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTemplateProjectAssignmenetCreate,
+		ReadContext:   resourceTemplateProjectAssignmentRead,
+		DeleteContext: resourceTemplateProjectAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:        schema.TypeString,
+				Description: "id of the template",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "id of the project",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func templateProjectAssignmentPayloadFromParameters(d *schema.ResourceData) client.TemplateAssignmentToProjectPayload {
+	result := client.TemplateAssignmentToProjectPayload{
+		ProjectId: d.Get("project_id").(string),
+	}
+
+	return result
+}
+
+func resourceTemplateProjectAssignmenetCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(*client.ApiClient)
+
+	templateId := d.Get("template_id").(string)
+	projectId := d.Get("project_id").(string)
+	request := templateProjectAssignmentPayloadFromParameters(d)
+	result, err := apiClient.AssignTemplateToProject(templateId, request)
+	if err != nil {
+		return diag.Errorf("could not assign template to project: %v", err)
+	}
+	resourceId := result.Id + "|" + projectId
+	d.SetId(resourceId)
+	return nil
+}
+
+func resourceTemplateProjectAssignmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(*client.ApiClient)
+
+	templateId := d.Get("template_id").(string)
+	template, err := apiClient.Template(templateId)
+	if err != nil {
+		return diag.Errorf("could not get template: %v", err)
+	}
+	var assignProjectId = d.Get("project_id").(string)
+	isProjectIdInTemplate:= false 
+	for _, projectId := range template.ProjectIds {
+		if assignProjectId == projectId {
+			isProjectIdInTemplate = true 
+		}
+	}
+	if !isProjectIdInTemplate {
+		return diag.Errorf("could not find projectId in template.\n projectId = %v, templateId = %v", assignProjectId, templateId)
+
+	}
```

Comment:
we don't also need to 

```
resourceId := result.Id + "|" + projectId
d.SetId(resourceId)
```

here?

Reply:
If I understand it correctly, we only need setId in Read while using terraform data. since data has not been implemented for this resource I don't think we need to set the Id .

in this case, Read function has been called by terraform after create function and the Id already set in create.